pages/actors/protocol.vue

```diff
@@ -1,68 +1,300 @@
+const validatorsList = computed(() => listsStore.validators);
 
-const listTableHeaders = [
-  {
-    key: "account",
-    label: "Name or Address",
-  },
-  { key: "list", label: "Role" },
-  { key: "timestamp", label: "Updated", sortable: true },
-];
+const combinedMintersList = computed(() => {
+  const combinedArray = mintersList.value.map((e) => {
+    const match = minters.find((f) => f.account === e.account);
+    const proposal = proposals.value.find(
+      (proposal) =>
+        proposal.executedEvent?.transactionHash === e.transactionHash
+    );
 
-const listsOptions = computed(() => [
-  ...new Set(lists.value.map((obj) => obj.list)),
-]);
+    return match ? { ...e, ...match, ...proposal } : e;
+  });
+  return combinedArray as any;
+});
 
-const selectedList = ref([]);
+const combinedValidatorsList = computed(() => {
+  const combinedArray = validatorsList.value.map((e) => {
+    const match = validators.find((f) => f.account === e.account);
+    const proposal = proposals.value.find(
+      (proposal) =>
+        proposal.executedEvent?.transactionHash === e.transactionHash
+    );
+    return match ? { ...e, ...match, ...proposal } : e;
+  });
+  return combinedArray as any;
+});
 
-const filteredLists = computed(() => {
+const filteredEarnerLists = computed(() => {
   const listsWithoutDuplicates = uniqBy(
     lists.value,
-    (item) => item.account + item.list,
-  );
-
-  if (selectedList.value.length === 0) return listsWithoutDuplicates;
+    (item) => item.account + item.list
+  )
+    .filter((item) => item.list === "earners")
+    .map((item) => {
+      const proposal = proposals.value.find(
+        (proposal) =>
+          proposal.executedEvent?.transactionHash === item.transactionHash
+      );
+      return { ...item, ...proposal };
+    });
 
-  return listsWithoutDuplicates.filter((obj) =>
-    selectedList.value.includes(obj.list),
-  );
+  return listsWithoutDuplicates;
 });
+
+const fetchEarnerClaimants = async () => {
+  try {
+    if (!listsStore.earners) return undefined;
+
+    const keys = listsStore.earners?.map((earner) =>
+      generateKeyEarnerClaimant(earner.account as Hash)
+    );
+
+    const claimants =
+      await apiStore.client.registrar!.protocolConfigs.getValuesByRawKeys(keys);
+
+    return listsStore.earners
+      .map((earner) => {
+        const key = generateKeyEarnerClaimant(earner.account as Hash);
+        const claimant = claimants.find((c: any) => c.key === key)?.value;
+        const proposal = proposals.value.find(
+          (proposal) =>
+            proposal.executedEvent?.transactionHash === earner.transactionHash
+        );
+
+        if (claimant) {
+          return {
+            earner: earner.account as Hash,
+            claimant: getAddress(trim(claimant as Hash)),
+            timestamp: earner.timestamp,
+            proposalId: proposal?.proposalId,
+            key,
+          };
+        }
+      })
+      .filter((item) => item);
```

Comment:
is it actually filtering anything here? did you mean something like

```suggestion
      .filter(Boolean);
```